File: SmartHomeTEC_API/Models/Certificate.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace SmartHomeTEC_API.Models
{
    [Table("Certificate")]
    public class Certificate
    {
        [Key]
        [ForeignKey("Device")]
        [Required]
        public required int SerialNumberDevice { get; set; }

        public string? Brand { get; set; }

        [Required]
        [ForeignKey("DeviceType")]
        public required string DeviceTypeName { get; set; }

        public string? ClientFullName { get; set; }

        [Required]
        public required string WarrantyStartDate { get; set; }

        public string? WarrantyEndDate { get; set; }

        [Required]
        [ForeignKey("Bill")]
        public required int BillNum { get; set; }

        [Required]
        [ForeignKey("Client")]
        [EmailAddress]
        public required string ClientEmail { get; set; }

        // Propiedades de Navegación:
        [JsonIgnore]
        public Bill? Bill { get; set; }

        [JsonIgnore]
        public Client? Client { get; set; }

        [JsonIgnore]
        public DeviceType? DeviceType { get; set; }

        [JsonIgnore]
        public Device? Device { get; set; }
    }
}

File: smart_home_tec_app/lib/JSONmodels/certificate.dart
import 'dart:convert';

Certificate certificateFromJson(String str) => Certificate.fromJson(json.decode(str));

String certificateToJson(Certificate data) => json.encode(data.toJson());

class Certificate {
  final int serialNumberDevice;
  final String brand;
  final String deviceTypeName;
  final String clientFullName;
  final String warrantyEndDate;
  final String warrantyStartDate;
  final int? billNum;
  final String clientEmail;

  Certificate({
    required this.serialNumberDevice,
    required this.brand,
    required this.deviceTypeName,
    required this.clientFullName,
    required this.warrantyEndDate,
    required this.warrantyStartDate,
    this.billNum,
    required this.clientEmail,
  });

  factory Certificate.fromJson(Map<String, dynamic> json) => Certificate(
        serialNumberDevice: json["serialNumberDevice"],
        brand: json["brand"],
        deviceTypeName: json["deviceTypeName"],
        clientFullName: json["clientFullName"],
        warrantyEndDate: json["warrantyEndDate"],
        warrantyStartDate: json["warrantyStartDate"],
        billNum: json["billNum"],
        clientEmail: json["clientEmail"],
      );

  Map<String, dynamic> toJson() => {
        "serialNumberDevice": serialNumberDevice,
        "brand": brand,
        "deviceTypeName": deviceTypeName,
        "clientFullName": clientFullName,
        "warrantyEndDate": warrantyEndDate,
        "warrantyStartDate": warrantyStartDate,
        "billNum": billNum,
        "clientEmail": clientEmail,
      };
}
